script/sign_upload_linux_pkgs.py

```diff
@@ -0,0 +1,243 @@
+            if channel in ['release']:
+                rpm_resign_cmd = os.path.join(repo_dir, "rpm-resign.exp")
+                cmd = "{} {} {} {}".format(rpm_resign_cmd, gpg_fingerprint, item, gpg_passphrase)
+                log_cmd = "{} {} {} {}".format(rpm_resign_cmd, gpg_fingerprint, item, 'NOTAREALPASSWORD')
+            else:
+                cmd = "rpmsign --resign --key-id={} {}".format(gpg_fingerprint, item)
+                log_cmd = cmd
+            logging.info("Running command: \"{}\"".format(log_cmd))
+
+            try:
+                subprocess.check_output(cmd, shell=True)
+                logging.info("RPM signing successful!")
+            except subprocess.CalledProcessError as cpe:
+                logging.error("Error running command: \"{}\"".format(log_cmd))
+                exit(1)
+
+    try:
+        os.chdir(repo_dir)
+        logging.debug('Changed directory to \"{}\"'.format(repo_dir))
+    except OSError as ose:
+        message = ('Error: could not change directory to {}: {}'.format(repo_dir, ose))
+        exit(message)
+
+    # Now upload to aptly and rpm repos
+
+    for item in ['upload_to_aptly', 'upload_to_rpm_repo']:
+        bucket = ''
+        if re.match(r'.*rpm.*', item):
+            bucket = 'brave-browser-rpm-staging-'
+        else:
+            bucket = 'brave-browser-apt-staging-'
+
+        upload_script = os.path.join(repo_dir, item)
+
+        TESTCHANNEL = 'test'
+
+        if s3_test_buckets:
+            upload_cmd = '{} {} {}'.format(upload_script, bucket + channel + '-' +
+                                           TESTCHANNEL, gpg_fingerprint)
+        else:
+            upload_cmd = '{} {} {}'.format(upload_script, bucket + channel, gpg_fingerprint)
+        logging.info("Running command: \"{}\"".format(upload_cmd))
+        try:
+            subprocess.check_output(upload_cmd, shell=True)
+        except subprocess.CalledProcessError as cpe:
+            logging.error("Error: {}".format(cpe))
+            exit(1)
+
+    # Not sure we need to change back to the original dir here,
+    # keeping it for now.
+    try:
+        os.chdir(saved_path)
+        logging.debug('Changed directory to \"{}\"'.format(saved_path))
+    except OSError as ose:
+        message = ('Error: could not change directory to {}: {}'.format(saved_path, ose))
+        exit(message)
+
+
+def download_from_github(args, logging):
+
+    file_list = []
+
+    # BRAVE_REPO defined in helpers.py
+    repo = GitHub(args.github_token).repos(BRAVE_REPO)
+    tag_name = args.tag
+    release = {}
+    releases = get_releases_by_tag(repo, tag_name, include_drafts=True)
+    if releases:
+        if len(releases) > 1:
+            exit("Error: More than 1 release exists with the tag: \'{}\'".format(tag_name))
+        release = releases[0]
+        logging.info("Searching for RPM/DEB packages in GitHub release: {}".format(release['url']))
+    for asset in release['assets']:
```

Comment:
Before iterating, should this also be checking if there are items present?
ex: `if release['assets'] is not None:`

Reply:
Done